python/beeai_framework/emitter/emitter.py

```diff
@@ -201,10 +203,13 @@ async def invoke(self, data: Any, event: EventMeta) -> None:
                 self.listeners.remove(listener)
 
             async def run(ln: Listener = listener) -> Any:
-                if inspect.iscoroutinefunction(ln.callback):
-                    return await ln.callback(data, event)
-                else:
-                    return ln.callback(data, event)
+                try:
+                    if inspect.iscoroutinefunction(ln.callback):
+                        return await ln.callback(data, event)
+                    else:
+                        return ln.callback(data, event)
+                except Exception as e:
+                    raise EmitterError.ensure(e, message="One of the provided Emitter callbacks has failed.")
```

Comment:
I think it would be worth adding context here, like which event callback errored. I can add that in #601 though if we want to just merge this first.

Reply:
how would you do that?